static/src/javascripts/bootstraps/common.js

```diff
@@ -241,6 +243,24 @@ define([
                 ['resize', 'scroll', 'orientationchange'].forEach(function (event) {
                     bean.on(window, event, mediator.emit.bind(mediator, 'window:' + event));
                 });
+                (function () {
+                    var running = false;
+                    var lastScrollY = window.pageYOffset;
+                    bean.on(window, 'scroll', function () {
+                        if (!running) {
+                            running = true;
+                            // fastdom is a handy replacement for simply throttling with
+                            // `requestAnimationFrame` (since we're calling `window.pageYOffset`)
+                            fastdom.read(function () {
+                                var scrollY = window.pageYOffset,
+                                    scrolledDown = lastScrollY < scrollY;
+                                lastScrollY = scrollY;
+                                mediator.emitEvent('window:throttledScroll', [scrollY, scrolledDown]);
+                                running = false;
```

Comment:
My question is why is this better than a centralised check?

As long as we're using FastDom, all reads will be batched?